src/main/java/org/apache/datasketches/QuantilesDoublesAPI.java

```diff
@@ -0,0 +1,325 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.datasketches;
+
+import static org.apache.datasketches.QuantileSearchCriteria.INCLUSIVE;
+
+public interface QuantilesDoublesAPI extends QuantilesAPI {
+
+  /**
+   * Same as {@link #getCDF(double[], QuantileSearchCriteria) getCDF(splitPoints, INCLUSIVE)}
+   * @param splitPoints an array of <i>m</i> unique, monotonically increasing quantiles
+   * @return a CDF array of m+1 double ranks (or probabilities) on the interval [0.0, 1.0).
+   */
+  default double[] getCDF(double[] splitPoints) {
+    return getCDF(splitPoints, INCLUSIVE);
+  }
+
+  /**
+   * Returns an approximation to the Cumulative Distribution Function (CDF) of the input stream
+   * as a monotonically increasing array of double ranks (or probabilities) on the interval [0.0, 1.0],
+   * given a set of splitPoints.
+   * The last rank in the returned array is always 1.0.
+   *
+   * <p>The resulting approximations have a probabilistic guarantee that can be obtained from the
+   * getNormalizedRankError(false) function.
+   *
+   * <p>If the sketch is empty this returns null.</p>
+   *
+   * @param splitPoints an array of <i>m</i> unique, monotonically increasing quantiles
+   * that divide the real number line into <i>m+1</i> consecutive non-overlapping intervals.
+   * It is not necessary to include either the minimum or maximum quantiles
+   * of the input stream in these split points.
+   *
+   * <p>If searchCrit is INCLUSIVE, the definition of an "interval" is
+   * exclusive of the left splitPoint and
+   * inclusive of the right splitPoint.</p>
+   *
+   * <p>If searchCrit is EXCLUSIVE, the definition of an "interval" is
+   * inclusive of the left splitPoint and
+   * exclusive of the right splitPoint.</p>
+   *
+   *<p>The left "splitPoint" for the lowest interval is the
+   * lowest quantile from the input stream retained by the sketch.
+   * The right "splitPoint" for the highest interval is the
+   * highest quantile from the input stream retained by the sketch.
+   * </p>
+   *
+   * @param searchCrit if INCLUSIVE, the weight of a given splitPoint quantile,
+   * if it also exists as retained quantile by the sketch,
+   * is included into the interval below.
+   * Otherwise, it is included into the interval above.
+   *
+   * @return an CDF array of m+1 double ranks (or probabilities) on the interval [0.0, 1.0).
+   */
+  double[] getCDF(double[] splitPoints, QuantileSearchCriteria searchCrit);
+
+  /**
+   * Returns the maximum quantile of the stream. This is provided for convenience, but is distinct from the largest
+   * quantile retained by the sketch algorithm.
+   *
+   * <p>If the sketch is empty this returns NaN.</p>
+   *
+   * @return the maximum quantile of the stream
+   */
+  double getMaxQuantile();
```

Comment:
Well it removes the ambiguity of "value", because ranks are also values.  I have also revised the web documentation to reflect this, which, in my mind makes it even more clear.  From the web doc:

"Note:
The term “value” can be ambiguous because items that we stream into a sketch are values and numeric ranks are also values. To avoid this ambiguity, we will use the term “quantiles” to refer to values that are streamed into a sketch even before they have been associated with a rank."

Reply:
I don't quite like the notion of the input domain consisting of quantiles

Reply:
Do you have a better suggestion?

Reply:
I would say that "item" is fine generic term for input data of an arbitrary type. At least I find it less awkward than calling all input pieces of data "quantiles".

Reply:
Now, for all of the Quantile sketches, SortedViews, SortedViewIterators, SketchIterators, the following method naming is used, where applicable:

> getMaxItem()
> getMinItem()
> getRank(quantile) //because it is the inverse of getQuantile(rank)

In addition, for the code & javadocs I tried to keep the naming consistent with the above.